adding javascript flag cleanup steps (#7011)
harness/developer-hub: docs/feature-flags/ff-creating-flag/manage-stale-flags.md

- Flags that have been globally set to a single value for longer than 30 days.

> #### Finding Potentially Stale Flags

Reviewer: ```suggestion
#### Finding potentially stale flags
```
We generally use lc for headings except for proper names etc.

> - At the top of your Feature Flags overview screen, you will find a list of filters.
> - One of those filters is labeled **Potentially Stale Flags**.
> - Click this list to see which flags have been highlighted by Harness as potentially stale.

Reviewer: ```suggestion
To see the flags identified by Harness as potentially stale, go to the top of the **Feature Flags Overview** and select the **Potentially Stale Flags** filter.
```
optional word-nerd sug.

- Java
- Kotlin
- Go
- Javascript 

There are some slight differences when using Javascript for flag cleanup automation compared to the other languages listed. Have a read to learn more about the differences in the implementation of flag cleanup automation in Javascript. 

### Set up a flag cleanup pipeline

>  1. In order to use flag cleanup automation you will need to [import a pipeline template](https://developer.harness.io/docs/platform/git-experience/import-a-template-from-git/). The GitHub repo to import from is [here](https://github.com/harness/flag_cleanup/blob/unscripted/docs/pipelines/flag_cleanup_pipeline.yaml). 

Reviewer: ```suggestion
 1. To use flag cleanup automation,  you must [import a pipeline template](https://developer.harness.io/docs/platform/git-experience/import-a-template-from-git/). The GitHub repo to import from is [here](https://github.com/harness/flag_cleanup/blob/unscripted/docs/pipelines/flag_cleanup_pipeline.yaml). 
```

- Note that the identifier and the name must match.

>  2. To configure secrets, here are the steps to do this:

Reviewer: ```suggestion
 2. To configure secrets:
```

>  - A Github connector to connect to the repo to clone down the code to cleanup. The [demo repo is here](https://github.com/harness/flag_cleanup) and the branch to use is “unscripted”.

Reviewer: ```suggestion
 - A Github connector to connect to the repo to clone the code to clean up. The [demo repo is here](https://github.com/harness/flag_cleanup) and the branch to use is “unscripted”.
```

 
 - A Docker connector to pull down the plugin image.
 
 ![Step four of importing a pipeline template](./static/setting-up-cleanup-4.png)

 ### Running the Flag Cleanup Pipeline

 1. Select the repository and branch for the codebase.
  
    ![Step one of running the flag cleanup pipeline](./static/running-cleanup-pipeline-1.png)

 2. Provide the GitHub token, GitHub username, and Harness API key.

    ![Step two of running the flag cleanup pipeline](./static/running-cleanup-pipeline-2.png)

 3. Specify the paths to the code (e.g., JavaScript paths).

    ![Step three of running the flag cleanup pipeline](./static/running-cleanup-pipeline-3.png)

 4. Select the DockerHub connector and run the pipeline.

    ![Step four of running the flag cleanup pipeline](./static/running-cleanup-pipeline-4.png)

As the pipeline runs, logs from the plugin show it getting the flags and performing the code changes.

![Output of cleanup pipline](./static/output-of-cleanup-pipeline.png)

You can then navigate to the PR and see the changes it made.

<DocVideo src="https://www.youtube.com/embed/sSP1nxrBwxo?si=dGI7vBmio6pfhWnX" />

### Understanding the JavaScript Flag Cleanup Tool

> This guide helps you understand the `properties.json` file structure, its role, and how to write custom flag cleanup rules for JavaScript.

Reviewer: ```suggestion
This section describes the `properties.json` file structure, its role, and how to write custom flag cleanup rules for JavaScript.
```
